### ISSUE TITLE
pre-pr-quality-gates: isServerSideProtectionHost reads .git/config as a path, so the law fails inside a git worktree

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#86]
`pre-pr-quality-gates` resolves the git host by reading `.git/config` as a filesystem path. Inside a `git worktree`, `.git` is a **file**, not a directory, so the read fails, the server-side-protection branch is never taken, and the law drops from a passing score to a failing one — for the same commit, in the same repository, on the same RuleOfCode version.

The effect is not a false finding about the code. It is a **verdict that depends on which directory you commit from**, which is worse: the gate says PASS in the primary checkout and FAIL in a worktree, and neither result is obviously the wrong one to the person reading it.

## Mechanism

`analyzeStatusChecks` is deliberately host-aware:

```js
hasRequiredStatusChecks:
  this.hasStatusCheckConfiguration(pipelineContent) ||
  (hasPipeline && this.isServerSideProtectionHost(projectRoot))
```

The second clause exists precisely so that hosts with server-side branch protection (Bitbucket, GitLab) are not punished for having no in-repo workflow files. But `isServerSideProtectionHost` implements the check as:

```js
readFile(join(projectRoot, '.git', 'config'), 'utf8')   // then: /bitbucket/ etc.
```

`join(projectRoot, '.git', 'config')` is only a real path when `.git` is a directory. In a linked worktree, `.git` is a regular file whose entire content is a pointer:

```
gitdir: /path/to/primary/.git/worktrees/<name>
```

so the read throws or returns nothing, the host is not recognised, and the law concludes there are no required status checks.

`analyzePRTemplates` in the same law already accepts `.bitbucket/pull_request_template.md`, so the law is not GitHub-only by design — this is one check falling back to a filesystem assumption the rest of the law does not make.

## Reproduce

```bash
git init primary && cd primary
git remote add origin https://bitbucket.org/example/example.git
# add any CI config the discovery step recognises, then:
npx ruleofcode audit --mode=pre-commit --staged      # pre-pr-quality-gates passes

git worktree add ../wt -b feature/x
cd ../wt
npx ruleofcode audit --mode=pre-commit --staged      # same commit, same version → fails
```

Observed on 7.19.0: the same law scored **80/100 (passed)** in the primary checkout and **45/100 (failed)** in a linked worktree of that same repository. The 35-point delta is exactly the required-status-checks deduction; the branch-protection deduction (−20) is present in both.

## Why this matters more than the score

Worktrees are the standard answer to "two agents/developers must not share one working directory" — they exist so concurrent work cannot corrupt a shared tree. A gate that changes its verdict when you adopt that isolation quietly penalises the safer setup, and it does so in a way that reads as a code problem rather than an environment one. Anyone hitting it will spend their time looking at their diff.

There is also a second-order risk in the passing direction: because the primary checkout satisfies the check through a filesystem coincidence rather than through anything about the repository's actual protection rules, a green result there is not evidence that server-side protection exists.

## Suggested fix

Ask git instead of reading the file:

```js
execFileSync('git', ['config', '--get', 'remote.origin.url'], { cwd: projectRoot })
```

This is correct in a primary checkout, a linked worktree, a submodule, and a repository whose config lives somewhere unusual. If shelling out is unwanted, resolve the pointer first: when `.git` is a file, parse the `gitdir:` line, walk up from `worktrees/<name>` to the common directory, and read `config` there.

The same assumption is worth grepping for elsewhere in the codebase — any law that touches `.git/<something>` by path has the same blind spot.

## Environment

- RuleOfCode 7.19.0
- Bitbucket remote, no `.github/` directory (the branch-protection sub-check is a separate matter and is not what this issue is about)
- Linked worktree created with `git worktree add`

**Comment by ruleofcodedev** [comment]
Fixed in #87, on `develop`.

Your framing decided the fix — "a verdict that depends on which directory you commit from" is the defect, not the score. Measured against a real worktree before and after:

| | before | after |
| --- | --- | --- |
| primary checkout | 35 | 35 |
| linked worktree | **0** | 35 |

The law asks git for the remote URLs now, so it is also correct in a submodule and wherever the config lives.

**You asked us to grep for the assumption elsewhere. There were six sites, and the second is worse than the one you hit:**

- `git-history-integrity` looked for a force-push guard in `.git/hooks/pre-push`. That misses the hook in **every husky project**, worktree or not, because husky moves `core.hooksPath` to `.husky` — and it drives a **violation**, so it accused repositories of having no force-push protection while their hook sat there working. RoC installs husky itself, so this was firing on projects that had done exactly what we told them to.
- `git-history-integrity` read `.git/config` for rebase policy.
- `merge-conflict-prevention` read `.git/config` for merge tools, and looked for a pre-merge hook under `.git/hooks`.
- `merge-conflict-prevention` read rebase state under `.git`. That one needed a distinction the helper did not have: an in-progress rebase is **per-worktree** state, while config and hooks are shared — asking the common directory would have reported the main checkout's rebase while judging yours.
- `GitHookPatterns.getHooksDirectory` joined the same path, with no callers.

Your second-order point is also right and is worth recording: the primary checkout was passing through a filesystem coincidence, not through anything about actual protection rules. A green result there was never evidence.

The helper existed already, from the previous time this was reported. That it did not cover these sites is the part we should have caught without a second report.
